audio/out/ao_oss.c

```diff
@@ -279,14 +279,21 @@ static int reopen_device(struct ao *ao, bool allow_format_changes)
     int format = ao->format;
     struct mp_chmap channels = ao->channels;
 
+    const char *device = PATH_DEV_DSP;
+    if (ao->device)
+        device = ao->device;
```

Comment:
Just randomly looked at this again. Sure that you don't need to check `ao->device[0]`? Or maybe that's intentional.